helpers/query.go

```diff
@@ -38,12 +38,17 @@ func QueryBalance(chain *relayer.Chain, address string, showDenoms bool) (sdk.Co
 
 	var out sdk.Coins
 	for _, c := range coins {
-		for _, d := range dts.DenomTraces {
-			switch {
-			case c.Amount.Equal(sdk.NewInt(0)):
-			case c.Denom == d.IBCDenom():
-				out = append(out, sdk.NewCoin(d.IBCDenom(), c.Amount))
-			default:
+		if c.Amount.Equal(sdk.NewInt(0)) {
+			continue
+		}
+
+		for i, d := range dts.DenomTraces {
```

Comment:
I think we should just do a query using [this route](https://github.com/cosmos/ibc-go/blob/2c38c0b230776c1cd0d5091be2935f3e3b36eecc/modules/apps/transfer/keeper/grpc_query.go#L20) instead of looping over all the traces